### DISCUSSION TITLE
Need to send response based on query parameter value

### DISCUSSION THREAD

**Discussion by javauser05** [mockoon/mockoon#1477] [Help] [answered]
Using version v7.0.0

I need to send response, based on value passed to query string as mentioned below.

1. For request **api/suppliers?status=active** wanted to send response as  **{ "messaage":"active" }**
Created below rule for it.

![image](https://github.com/user-attachments/assets/c907cc79-429f-43ad-8b39-9df1e86f3828)

2. For **api/suppliers?status=inactive** wanted to send response as  **{ "messaage":"inactive" }**  
Created below rule for it.

![image](https://github.com/user-attachments/assets/b0cda212-0c31-423a-bed8-05d1edd83873)

However, when I test this, for **/api/suppliers?status=active** and **/api/suppliers?status=inactive**, I get response as **{ "messaage":"active" }** only .

Attached **APIMock.json** for reference.
Please help what I am missing.

[APIMock.json](https://github.com/user-attachments/files/16460976/APIMock.json)

**Comment by 255kb** [comment, marked as answer]
In your rules, you used the "route params" instead of the "query string". 
Everything after the `?` is part of the query string. Route parameters are for parameters that are part of the path: `/path/:param1/:param2`.

Also, the rule does not support templating yet. It will be added in v8.4.0. 

The corrected rules should be: 

![image](https://github.com/user-attachments/assets/ec732807-8c88-438e-b4df-f7c1f38e69f8)
![image](https://github.com/user-attachments/assets/ef709a88-3c29-4304-a113-a738235173f4)

**Reply by javauser05** [reply]
It works. Thanks a lot @255kb
